drivers/flash/soc_flash_mcux.c

```diff
@@ -39,6 +39,11 @@ LOG_MODULE_REGISTER(flash_mcux);
 
 #define SOC_NV_FLASH_NODE DT_INST(0, soc_nv_flash)
 
+
+#ifdef CONFIG_FLASH_MCUX_LPC_IAP_WRITE_BUFFER
+static uint8_t mcux_flash_write_buffer[DT_PROP(SOC_NV_FLASH_NODE, write_block_size)][2];
```

Comment:
`write_block_size` is set to 1 when `CONFIG_FLASH_MCUX_LPC_IAP_WRITE_BUFFER` is defined. Don't need to read the property here.

Reply:
`write_block_size` is kept at 512 in the DTS definition, so this will still allocate correctly sized blocks. `CONFIG_FLASH_MCUX_LPC_IAP_WRITE_BUFFER ` only changes the recorded value for `write_block_size` definition within the driver. So the DTS still defines a `write_block_size` of 512, but the driver will report a write block size of 1 to any API consumers. This unaligned write logic will handle properly writing data with alignment less than 512 bytes.